### PR TITLE
Fix #11018

### LINKED ISSUE #11018: Casting a seq to another seq generates invalid code with --newruntime
Casting a seq or a string to another seq generates invalid C code with --newruntime

### A couple of examples:
seq -> seq

```nim
discard cast[seq[uint8]](@[1])
```
string -> seq
```nim
discard cast[seq[uint8]]("test")
```

### Current Output
```
Error: execution of an external compiler program 'gcc -c  -w  -I/home/clyybber/builds/nim/lib -I/home/clyybber/projects/tests -o /home/clyybber/.cache/nim/convrt_d/convrt.c.o /home/clyybber/.cache/nim/convrt_d/convrt.c' failed with exit code: 1

/home/clyybber/.cache/nim/convrt_d/convrt.c: In function ‘NimMainModule’:
/home/clyybber/.cache/nim/convrt_d/convrt.c:157:2: error: conversion to non-scalar type requested
  (void)(((tySequence_6H5Oh5UUvVCLiakt9aTwtUQ) (TM_KUSKOxFeGyNT8JCQmxu3cw_2)));
  ^
```

### Expected Output

Compile successfully

### Additional Information
Casting an array to a seq does not generate invalid code:
```nim
discard cast[seq[uint8]]([1])
```

```
$ nim -v
Nim Compiler Version 0.19.9
```

### PR DESCRIPTION
**Pull request by Clyybber**
This is probably total bollocks, but here's my attempt at fixing #11018.
Appreciate advice on how to fix it correctly (if this is wrong)  :)

### CODE REVIEW

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -1839,6 +1839,9 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
     if srcTyp.kind in {tyPtr, tyPointer} and etyp.kind in IntegralTypes:
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
+    elif etyp.kind == tySequence:
```
Comment (Araq) [inline compiler/ccgexprs.nim]: Please try

``elif p.config.selectedGc == gcDestructors and etyp.kind == tySequence:``

Comment (Clyybber) [reply compiler/ccgexprs.nim]: Ah damn. Yeah I forgot that.

**Comment by Clyybber** [comment]
This patch is incorrect for sure, because it's missing casting the length of the sequence.

File: compiler/ccgexprs.nim [hunk 2]
```diff
@@ -1839,6 +1839,9 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
     if srcTyp.kind in {tyPtr, tyPointer} and etyp.kind in IntegralTypes:
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
+    elif p.config.selectedGc == gcDestructors and etyp.kind == tySequence:
+      putIntoDest(p, d, e, "(($1_Content*) ($2.p))" %
```
Comment (Araq) [inline compiler/ccgexprs.nim]: Sorry, thinking about this more, this is still incorrect? The cast's result should still be (length, pointer) pair and not just the casted pointer.

Comment (Clyybber) [reply compiler/ccgexprs.nim]: Yeah, (as I pointed out above). I'm working on a correct fix.

**Pull request merged**